src/cc/syms.h

```diff
@@ -147,3 +147,47 @@ class ProcSyms : SymbolCache {
   virtual bool resolve_name(const char *module, const char *name,
                             uint64_t *addr);
 };
+
+class BuildSyms {
+  struct Symbol {
+    Symbol(const std::string *name, uint64_t start, uint64_t size)
+      :name(name), start(start), size(size) {}
+    const std::string *name;
+    uint64_t start;
+    uint64_t size;
+
+    bool operator<(const struct Symbol &rhs) const {
+      return start < rhs.start;
+    }
+  };
+
+  struct Module {
+    Module(const char *module_name):
+      module_name_(module_name),
+      loaded_(false) {}
+    const std::string module_name_;
+    const std::string build_id_;
+    bool loaded_;
+    std::unordered_set<std::string> symnames_;
+    std::vector<Symbol> syms_;
+    bcc_symbol_option symbol_option_;
+
+    bool load_sym_table();
+    static int _add_symbol(const char *symname, uint64_t start, uint64_t size,
+                            void *p);
+    bool resolve_addr(uint64_t offset, struct bcc_symbol*, bool demangle=true);
+  };
+
+  std::unordered_map<std::string, Module*> buildmap_;
+
+public:
+  BuildSyms() {}
+  virtual ~BuildSyms() {
+    std::unordered_map<std::string, Module*>::iterator it;
+    for (it=buildmap_.begin();it!=buildmap_.end();++it) {
+      delete it->second;
```

Comment:
Why not just `unique_ptr`?

Reply:
Did you mean buildmap_ can be an unique_ptr ?

Reply:
I mean the map values could be, so you don't have to remember manually delete them?